examples/preprocessing/plot_pace_by_data_perc.py

```diff
@@ -0,0 +1,225 @@
+    max_val: int = 226,
+    random_state: int | None = None,
+) -> NDArrayInt:
+    """
+    Calculate the number of points of each observation.
+
+    Calculate the number of points of each observation to transform a regular
+    dataset into an irregular one, based on a percentage of measurement from
+    the whole dataset.
+
+    Args:
+        percentage: percentage of measurements to retain.
+        size: number of observations.
+        min_val: minimum number of measurements to retain per observations.
+        max_val: number of measurements per observation.
+        random_state: random state.
+
+    Returns:
+        Array of integers with the number of time points to retain.
+    """
+    target_sum = round((percentage / 100) * size * max_val)
+
+    base = np.full(size, min_val)
+    remainder = target_sum - base.sum()
+
+    rng = np.random.default_rng(random_state)
+
+    if remainder > 0:
+        increments = rng.multinomial(remainder, [1/size] * size)
+        result = base + increments
+    else:
+        result = base.copy()
+
+    return result.tolist()
+
+# %%
+# In order to generate sufficient data for our analysis, we will perform the
+# PACE algorithm for each of the different data percentages: 10, 20 and 35.
+# Our interest resides in analysing the principal components, full
+# reconstructions of the data and calculating the average mean squared error
+# of the reconstructed curves, compared to the original ones.
+n_components = 3
+components_all = []
+reconstructed_all = []
+mse_all = []
+
+random_state = 11
+data_percentages = np.array([5, 15, 25])
+
+for perc in data_percentages:
+    measurements_per_obs = generate_num_measurements(
+        perc,
+        random_state=random_state,
+    )
+
+    irregular_canadian = irregular_sample(
+        canadian,
+        measurements_per_obs,
+        random_state=random_state,
+    )
+    irregular_canadian.coordinate_names = canadian.coordinate_names
+    irregular_canadian.argument_names = canadian.argument_names
+
+    pace = PACE(
+        n_components=n_components,
+        bandwidth_mean=np.array([0.1, 10]),
+        bandwidth_cov=10.0,
+        n_grid_points=25,
+    )
+    pace_scores = pace.fit_transform(irregular_canadian)
+    reconstructed_pace = pace.inverse_transform(pace_scores)
+    components_all.append(pace.components_)
+    reconstructed_all.append(reconstructed_pace)
+
+    print(pace.explained_variance_ratio_[:8])
+
+    grid1 = canadian.grid_points[0]
+    grid2 = reconstructed_pace.grid_points[0]
+
+    common_grid = np.intersect1d(grid1, grid2)
+    idx1 = np.where(np.isin(grid1, common_grid))[0]
+    idx2 = np.where(np.isin(grid2, common_grid))[0]
+
+    aligned_true = canadian.data_matrix[:, idx1]
+    aligned_pred = reconstructed_pace.data_matrix[:, idx2]
+
+    mse_per_curve = np.mean((aligned_true - aligned_pred) ** 2, axis=1)
+    average_mse = np.mean(mse_per_curve)
+
+    mse_all.append(average_mse)
+
+print(mse_all)
```

Comment:
Maybe put it into a dataframe and let it be the last sentence, so that it is printed with labels and could be understood.